### PR TITLE
ref(types): Extend type annotations

### PR DESCRIPTION
**Pull request by tkaemming**
This adds several low-hanging fruit type annotations, mostly around the command line interface.

### CODE REVIEW

File: snuba/cli/bulk_load.py [hunk 1]
```diff
@@ -10,15 +11,24 @@
...
 @click.option("--dest-table", help="Clickhouse destination table.")
 @click.option("--log-level", default=settings.LOG_LEVEL, help="Logging level to use.")
-def bulk_load(dataset, dest_table, source, log_level):
+def bulk_load(
+    *,
+    dataset_name: Optional[str],
+    dest_table: Optional[str],
+    source: Optional[str],
+    log_level: str,
+) -> None:
```
Comment (tkaemming) [inline snuba/cli/bulk_load.py]: These types reflect the current option behavior, but do cause these new type errors:

```
snuba/cli/bulk_load.py:43: error: Argument 1 to "get_dataset" has incompatible type "Optional[str]"; expected "str"
snuba/cli/bulk_load.py:47: error: Argument "path" to "load" of "PostgresSnapshot" has incompatible type "Optional[str]"; expected "str"
```

File: snuba/cli/cleanup.py [hunk 2]
```diff
@@ -27,20 +27,29 @@
 @click.option("--database", default="default", help="Name of the database to target.")
 @click.option(
     "--dataset",
+    "dataset_name",
```
Comment (fpacifici) [inline snuba/cli/cleanup.py]: nice, I was looking for a way to do that when I refactored the consumer.

File: snuba/cli/bulk_load.py (hunk 1 above)
Comment (fpacifici) [reply snuba/cli/bulk_load.py]: in the confirm_load script the option is defined in the same way but you provided str not Optional[str]. Is that intentional ?

File: snuba/cli/consumer.py [hunk 3]
```diff
@@ -91,23 +95,24 @@
...
-    dogstatsd_port,
-    stateful_consumer,
-):
+    *,
+    raw_events_topic: Optional[str],
+    replacements_topic: Optional[str],
+    commit_log_topic: Optional[str],
+    control_topic: Optional[str],
+    consumer_group: str,
+    bootstrap_server: Sequence[str],
```
Comment (fpacifici) [inline snuba/cli/consumer.py]: This has None as default.

File: snuba/cli/consumer.py [hunk 4]
```diff
@@ -91,23 +95,24 @@
...
-    dogstatsd_host,
-    dogstatsd_port,
-    stateful_consumer,
-):
+    *,
+    raw_events_topic: Optional[str],
+    replacements_topic: Optional[str],
+    commit_log_topic: Optional[str],
+    control_topic: Optional[str],
+    consumer_group: str,
```
Comment (fpacifici) [inline snuba/cli/consumer.py]: Isn't this Optional as well ?

**Review by fpacifici — COMMENTED** [review]
A couple of questions on a few  Optional parameters otherwise it is good.

File: snuba/cli/bulk_load.py (hunk 1 above)
Comment (tkaemming) [reply snuba/cli/bulk_load.py]: That was an oversight on my part: this file (`bulk_load`) is correct, the other one (`confirm_load`) should be fixed to mirror this approach.

File: snuba/cli/consumer.py (hunk 3 above)
Comment (tkaemming) [reply snuba/cli/consumer.py]: Removed the unused default: 0133b2dc60cbc37e9d0e53508321150f2f678405

File: snuba/cli/bulk_load.py (hunk 1 above)
Comment (tkaemming) [reply snuba/cli/bulk_load.py]: Corrected the optionality of the function arguments here: ba4d0e05e0475badfae2f473ed6352481b937460

**Review by fpacifici — APPROVED** [review]

**Pull request merged**